buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java

```diff
@@ -1095,12 +1096,12 @@ private void createConfiguration() {
             );
         }
         // Make sure no duplicate config keys
-        settings.keySet().stream().filter(OVERRIDABLE_SETTINGS::contains).forEach(defaultConfig::remove);
+        settings.keySet().stream().filter(OVERRIDABLE_SETTINGS::contains).forEach(baseConfig::remove);
 
         try {
             Files.write(
                 configFile,
-                Stream.concat(settings.entrySet().stream(), defaultConfig.entrySet().stream())
+                Stream.concat(settings.entrySet().stream(), baseConfig.entrySet().stream())
```

Comment:
I'm a bit confused. Is the configuration in `defaultConfig` not needed? `ElasticsearchCluster.commonNodeConfig()` places a bunch of stuff in there. Won't this change now no longer write those config items to the config file or am I missing something?

Reply:
On [line 1052](https://github.com/elastic/elasticsearch/pull/55560/files#diff-a704b6e9e08a9ccfcc05e2ae59844495R1052) we initialize `baseConfig` with `defaultConfig`:
`Map<String, String> baseConfig = new HashMap<>(defaultConfig);`

Reply:
That's indeed what I was missing.